### PR TITLE
Revert "Have EntityCommands methods consume self for easier chaining"

### LINKED ISSUE #15521: Consider reverting change to EntityCommands calling convention
The addition of `insert_if` is indeed nice. I was commenting purely on methods now consuming `self`. I realize now that I wasn't clear about that in my earlier comments, sorry about that

Within the changes in this PR I counted 6 cases where the change to consuming `self` enabled chaining.
On the other hand I counted 15 instances where doing re-assignment (`entity_commands = entity_commands.XXX()`) was now necessary.

IMO, that's not looking great.

> I'd like to keep the API in place for the RC, see how the ecosystem at large reacts to it

Right, we do have release candidates now. That does ease my worries a bit.

_Originally posted by @tim-blackbird in https://github.com/bevyengine/bevy/issues/14897#issuecomment-2381336342_

### PR DESCRIPTION
**Pull request by tim-blackbird**
As discussed in #15521

- Partial revert of #14897, reverting the change to the methods to consume `self`
- The `insert_if` method is kept

The migration guide of #14897 should be removed
Closes #15521

### CODE REVIEW

File: crates/bevy_ecs/src/system/commands/mod.rs
Comment (tim-blackbird) [inline crates/bevy_ecs/src/system/commands/mod.rs:1408]: #14897 mistakenly made `despawn` return self

File: examples/3d/motion_blur.rs
Comment (tim-blackbird) [inline examples/3d/motion_blur.rs:159]: Only indentation/whitespace changes below this comment in this file :)

gits diffs do a surprisingly bad job with indentation changes

**Comment by MrGVSV** [comment]
Could you provide details as to why this revert should be merged in your PR description?

File: crates/bevy_ecs/src/system/commands/mod.rs
Comment (tim-blackbird) [inline crates/bevy_ecs/src/system/commands/mod.rs:395]: IMO the only clear win for consuming `self`, but that's because the spawn methods return `EntityCommands` by value either way

**Comment by tim-blackbird** [comment]
> Could you provide details as to why this revert should be merged in your PR description?

Apologies, I forgot to link to the relevant issue. Added to the PR description.

**Comment by Luminoth** [comment]
I'm hoping the issue in question gets a little more time to discuss before this goes through.

File: crates/bevy_ecs/src/system/commands/mod.rs
Comment (Luminoth) [reply crates/bevy_ecs/src/system/commands/mod.rs:1408]: For context, this wasn't a mistake, it was something acknowledged in the PR: https://github.com/bevyengine/bevy/pull/14897#issuecomment-2307615139

**Comment by alice-i-cecile** [comment]
Yep, I intend to leave this discussion open in the linked issue until the release candidate is due to be cut. Given the level of controversy and breakage, I don't think it should have been merged without more discussion, so the default stance here is "revert".

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_ecs/src/system/commands/mod.rs
Comment (chescock) [inline crates/bevy_ecs/src/system/commands/mod.rs:1430]: This `allow` could have been removed when the method was renamed from `add` to `queue`.  That doesn't need to be part of this PR, but I noticed it because it was added in the PR you're reverting.

File: crates/bevy_ecs/src/system/commands/mod.rs
Comment (chescock) [inline crates/bevy_ecs/src/system/commands/mod.rs:1520]: I think this makes sense, but I want to highlight that the changes to `EntityEntryCommands` aren't exactly a revert.  This type didn't exist at the time of the original change, and it *could* make sense for it to work differently from `EntityCommands`.

**Review by chescock — APPROVED** [review]
I compared the diff to the one from #14897, and this does revert the `&mut self` to `self` changes while preserving the `insert_if` changes.  (The changes to `EntityEntryCommands` aren't exactly a revert, but seem to be in the same spirit.)

File: crates/bevy_ecs/src/system/commands/mod.rs
Comment (alice-i-cecile) [reply crates/bevy_ecs/src/system/commands/mod.rs:1430]:
```suggestion
```

**Comment by alice-i-cecile** [comment]
@tim-blackbird I'll merge this once you resolve the merge conflicts :)

File: crates/bevy_ecs/src/system/commands/mod.rs
Comment (tim-blackbird) [reply crates/bevy_ecs/src/system/commands/mod.rs:1519]: That is true, I should have thought to mention this. Sorry about that

**Pull request merged**
